content/blog/fast-data-processing-pipeline-for-predicting-flight-delays-using-apache-apis-kafka-spark-streaming-and-machine-learning-part-3.md

```diff
@@ -0,0 +1,295 @@
+
+According to Bob Renner, CEO of [Liaison Technologies](https://www.liaison.com/), the possibility to blend machine learning with real-time transactional data flowing through a single platform is opening a world of new possibilities, such as enabling organizations to take advantage of opportunities as they arise. According to [Gartner](https://www.gartner.com/newsroom/id/3812063) Over the next few years, virtually every app, application and service will incorporate some level of machine learning. Leveraging these opportunities requires fast and scalable data processing pipelines.
+
+![MapR Platform](/img/mapr-platform.png)
+
+This is the third in a series of blogs, which discusses the architecture of a data pipeline that combines streaming data with machine learning and fast storage. [The first post](https://developer.hpe.com/blog/fast-data-processing-pipeline-for-predicting-flight-delays-using-apache-/) discussed creating a machine learning model to predict flight delays. The [second post](https://developer.hpe.com/blog/fast-data-processing-pipeline-for-predicting-flight-delays-using-apache-apis-kafka-spark-streaming-and-machine-learning-part-2/) discussed using the saved model with streaming data to do real-time analysis of flight delays. This third post will discuss fast storage and analysis with MapR Database, Apache Spark, Apache Drill and OJAI.
+
+**Machine Learning Logistics and Data Pipelines**
+
+Machine Learning usually refers to the model training piece of an ML workflow. But, as Ted Dunning says, 90% of the effort around Machine Learning is data logistics which includes all of the aspects that occur before and after this training.  When you combine event streams with microservices, you can greatly enhance the agility with which you build, deploy, and maintain complex data pipelines. Pipelines are constructed by chaining together microservices, each of which listens for the arrival of some data, performs its designated task, and optionally publishes its own messages to another topic.  Combining event-driven data pipelines with machine learning can handle the logistics of machine learning in a flexible way by:
+
+* Making input and output data available to independent consumers
+* Managing and evaluating multiple models and easily deploying new models
+* Monitoring and analyzing models, with historical and real-time data.
+
+![MapR Architectures](/img/mapr-architectures.png)
+
+Architectures for these types of applications are discussed in more detail in the eBooks: Machine Learning Logistics, Streaming Architecture, and Microservices and Containers.
+
+The following figure depicts the (simplified) data pipeline for this tutorial:
+
+* Flight trip data is published to a MapR Event Streams (ES) topic using the Kafka API. (Note this data contains the actual delay label, in the real world architecture the actual delay label would come later in a different topic, but to keep the tutorial code simple it is combined with the input data).
```

Comment:
Put a period after label. Add a hyphen between “real” and “world”. Add a comma after “architecture” so that the second sentence reads: “In the real-world architecture, the actual…”

![image](https://user-images.githubusercontent.com/77496261/137183109-ccc81fb8-9f64-4117-9455-c09e661c0b9d.png)

Reply:
done